sim/validator.py

```diff
@@ -6,209 +6,149 @@
-            results_path (str): path to results - if None defaults to output_path
-            simulated_reads_path (str): path to simualted reads
-            exist_ok (bool): Whether or not to throw an error if a results directory already exists
-            branch (str): Checkout git branch on the repo (default None)
-            light_mode (bool): switch to delete non-essential analysis files (default False)
+            light_mode (bool): keep only output statistics to save disk space (default False)
 
         Returns:
             None
     """
-
-    # Add trailing slash
-    btb_seq_path = os.path.join(btb_seq_path, '')
-    output_path = os.path.join(output_path, '')
-    if not results_path:
-        results_path = output_path
-    else:
-        results_path = os.path.join(results_path, '')
-
-    # Run simulations if path to simulated reads not provided 
-    if not simulated_reads_path:
-        simulated_reads_path = simulate(output_path, samples)     
-    elif not os.path.isdir(output_path):
-        raise Exception("Output path not found")
-
-    # Validate btb_seq_path
-    if not os.path.isdir(btb_seq_path):
-        raise Exception("Pipeline code repository not found")
-
-    os.makedirs(results_path, exist_ok=exist_ok)
-    btb_seq_backup_path = results_path + '/btb-seq/'
-    btb_seq_results_path = results_path + '/btb-seq-results/'
-    site_stats_path = results_path + '/site-stats/'
-
-    # Paths to mask file and simulated reads directory 
-    mask_filepath = btb_seq_backup_path + "references/Mycbovis-2122-97_LT708304.fas.rpt.regions"
-
-    # TODO: handle dwgsim vcf files. Make sure we are taking into account variants it might generate
-
-    # Create Output Directories
-    os.makedirs(btb_seq_results_path, exist_ok=exist_ok)
-    os.makedirs(site_stats_path, exist_ok=exist_ok)
-
-    # Backup btb-seq code
-    # TODO: exclude the work/ subdirectory from this operation.
-    #   This could potentially copy large amounts of data
-    #   from the work/ directory nextflow generates
+    # Define output paths
+    genomes_path = os.path.join(output_path, 'genomes')
+    reads_path = os.path.join(output_path, 'reads')
+    btb_seq_backup_path = os.path.join(output_path, 'btb-seq')
+    results_path = os.path.join(output_path, 'sequenced')
+    stats_path = os.path.join(output_path, 'stats')
+
+    # Initialise
+    # TODO: exclude the work/ subdirectory from this operation to save space
+    os.makedirs(output_path)
     shutil.copytree(btb_seq_path, btb_seq_backup_path, symlinks=True)
+    os.makedirs(stats_path, exist_ok=False)
 
-    # TODO: Use nextflow's method of choosing github branches
-    #    the method handles automatic pulling of github branches.
-    if branch:
-        checkout(btb_seq_backup_path, branch)
+    # Simulate
+    simulated_samples = simulate(samples, genomes_path, reads_path)
+    results_path = sequence(btb_seq_path, reads_path, results_path)
```

Comment:
Slightly confused by this rn -> results_path already set on line 88.

Reply:
Ok i see, because it's appended with` /Results...` in` sequence()`. Maybe it could be a new variable. Maybe something like `results_path = sequence(btb_seq_path, reads_path, results_path_parent)`. This would be more functional. Very minor point though ...

Reply:
Add comment above line 99 `# Sequence`

Reply:
good idea